src/app/ReadClient.cpp

```diff
@@ -427,6 +429,31 @@ CHIP_ERROR ReadClient::GenerateDataVersionFilterList(DataVersionFilterIBs::Build
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ReadClient::OnActiveModeNotification()
+{
+    VerifyOrReturnError(IsSubscriptionType() && IsIdle(), CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(mPeerActivePeriod != System::Clock::kZero, CHIP_ERROR_INCORRECT_STATE);
+    // TODO: Add Better the error code to inform the ActiveModeWakeUp and trigger the subscription retry
+    Close(CHIP_ERROR_TIMEOUT);
```

Comment:
This is missing a bunch of the logic a normal subscription timeout has.  Why?

Reply:
I realized that I have some misunderstanding for some parts of the icd. Updated.